### DISCUSSION TITLE
Typescript for functions is a pointless time sink. Change my mind?

### DISCUSSION THREAD

**Discussion by 0x62** [supabase/supabase#10763] [Polls]
I've been using Supabase for a while now, and overall I love it as a platform. In the past I've also used Hasura, and the UI and developer experience here is great, except for one thing. Typescript.

When I'm creating a Function, 99% of the time it's a basic interaction with a 3rd party API. Usually there aren't Deno libraries, so it's either spend time defining a bunch of types which are only used once, or `Record<string, any>` and be done with it.

I tried really hard to embrace it, but in the end I spent more time fixing Typescript issues than I did actually getting stuff done. I'd be curious to hear what other people think?

**Comment by iAmWillShepherd** [comment]
I love Typescript, and I agree that requiring Typescript for functions is not worth the effort. 

Here's what I have to deal with...
<img width="2228" alt="Screen Shot 2022-12-11 at 7 54 55 PM" src="https://user-images.githubusercontent.com/1715082/206944578-31670165-892b-4c8a-bffb-19d141c13a4a.png">

This doesn't even get into the fact that half the time, the library I'm using doesn't even include types (or doesn't exist on definitely typed). I'm not sure how much of this is due to the decision to use Deno, but so far, I'm not enjoying the DX of the decision. I'm hoping it substantially improves in the future.

**Reply by thorwebdev** [reply]
@iAmWillShepherd it looks like your IDE might not properly set up for usage with Deno. Can you follow the steps here: https://deno.land/manual/getting_started/setup_your_environment ?

If you're using vscode you can follow this video guide also: https://www.youtube.com/watch?v=jZgyOJGWayQ&list=PL5S4mPUpp4OulD3olUW8Eq1IYKpUbk5Ob&t=27s

**Comment by sosensible** [comment]
I don't believe the answer is black or white, hot or cold, or any other binary reality. You seemed to agree because you expressed 99% of the time. Of course, that implies you have written a hundred or more functions. If so, congrats on the productivity.

So, it seems the question is can we make it optional as Vue does? In the script tag in Vue, we put `lang="ts"` or it defaults to `js`. In this case, it would default to `ts` to avoid breaking changes with legacy functions. Would that resolve your challenges?

P.S.
I am also aware that failing code impacts all of us in system load, and thereby the costs. It would be interesting to discover the trade-offs for doing non-typescript.
